kiali-server/templates/role-viewer.yaml

```diff
@@ -1,3 +1,4 @@
+{{- if or (.Values.deployment.view_only_mode) (ne .Values.auth.strategy "anonymous") -}}
```

Comment:
anonymous mode can still be used with view_only_mode=true (in fact, I think many people use it that way).

I don't think auth.strategy setting should affect which role is created.

Can you explain the thinking because checking for anonymous mode?

Reply:
I get your point. 
The goal is to have the same behavior with the [roleBinding](https://github.com/kiali/helm-charts/blob/cb4aebc46dc90ed0daa766c30777632bbd61fd27/kiali-server/templates/rolebinding.yaml#L5).
So if the roleBinding is set to viewer role, the viewer role is created (same for full rights role)

Reply:
Ah, right. I remember this now.

* https://github.com/kiali/kiali/issues/5611.